server/test/com/xsn/explorer/data/BlockPostgresDataHandlerSpec.scala

```diff
@@ -232,4 +256,13 @@ class BlockPostgresDataHandlerSpec extends PostgresDataHandlerSpec with BeforeAn
     result.difficulty mustEqual expected.difficulty
     result.nonce mustEqual expected.nonce
   }
+
+  private def matches(expected: BlockHeader, result: BlockHeader) = {
+    // NOTE: transactions and confirmations are not matched intentionally
+    result.hash mustEqual expected.hash
+    result.previousBlockhash mustEqual expected.previousBlockhash
+    result.merkleRoot mustEqual expected.merkleRoot
+    result.height mustEqual expected.height
+    result.time mustEqual expected.time
```

Comment:
This doesn't apply here, you are not checking whether the filter is present, a way would be:
```
(expected, result) match {
  case (e: BlockHeader.HasFilter, r: BlockHeader.HasFilter) => matchFilter(e.filter, r.filter)
  case _ => ()
}
```